src/apps/carbon-defi/ethereum/carbon-defi.strategy.contract-position-fetcher.ts

```diff
@@ -0,0 +1,160 @@
+import { ETH_ADDR_ALIAS, ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { drillBalance } from '~app-toolkit/helpers/drill-balance.helper';
+import { ContractPositionBalance, RawContractPositionBalance } from '~position/position-balance.interface';
+import { MetaType } from '~position/position.interface';
+import {
+  DefaultContractPositionDefinition,
+  UnderlyingTokenDefinition,
+  GetTokenDefinitionsParams,
+  GetDisplayPropsParams,
+  GetDataPropsParams,
+} from '~position/template/contract-position.template.types';
+import { CustomContractPositionTemplatePositionFetcher } from '~position/template/custom-contract-position.template.position-fetcher';
+import { Network } from '~types';
+
+import { CarbonDefiViemContractFactory } from '../contracts';
+import { CarbonController } from '../contracts/viem/CarbonController';
+
+interface Order {
+  y: bigint;
+  z: bigint;
+  A: bigint;
+  B: bigint;
+}
+
+interface Strategy {
+  id: bigint;
+  owner: string;
+  tokens: readonly [string, string];
+  orders: readonly [Order, Order];
+}
+
+type StrategyDefinition = DefaultContractPositionDefinition & {
+  strategy: Strategy;
+};
+
+type StrategyProps = {
+  strategy: Strategy;
+};
+
+function isActiveStrategy(strategy: Strategy) {
+  const [buy, sell] = strategy.orders;
+  return !!buy.A || !!buy.B || !!sell.A || !!sell.B;
+}
+
+@PositionTemplate()
+export class EthereumCarbonDefiStrategyContractPositionFetcher extends CustomContractPositionTemplatePositionFetcher<
+  CarbonController,
+  StrategyProps,
+  StrategyDefinition
+> {
+  groupLabel = 'Carbon Defi';
+
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(CarbonDefiViemContractFactory) protected readonly factory: CarbonDefiViemContractFactory,
+  ) {
+    super(appToolkit);
+  }
+
+  getContract() {
+    return this.factory.carbonController({
+      address: '0xc537e898cd774e2dcba3b14ea6f34c93d5ea45e1',
+      network: Network.ETHEREUM_MAINNET,
+    });
+  }
+
+  override async getDataProps(params: GetDataPropsParams<CarbonController, StrategyProps, StrategyDefinition>) {
+    return { strategy: params.definition.strategy };
+  }
+
+  async getLabel(params: GetDisplayPropsParams<CarbonController>): Promise<string> {
+    const tokens = params.contractPosition.tokens;
+    return `${tokens[0].symbol} / ${tokens[1].symbol}`;
+  }
+
+  async getDefinitions(): Promise<StrategyDefinition[]> {
+    const contract = this.getContract();
+    const address = contract.address;
+    const pairs = await contract.read.pairs();
+    const getStrategies = pairs.map(pair => contract.read.strategiesByPair([...pair, BigInt(0), BigInt(0)]));
+    const strategies = await Promise.all(getStrategies).then(matrix => matrix.flat());
+    return strategies.filter(isActiveStrategy).map(strategy => ({ address, strategy: strategy }));
```

Comment:
Essentially, the balances are stil propagated from the definition all the way to dataProps and then balances. The definition should not contain balances

Reply:
Basically getting the definitions (strategies in Carbon) from the contract always returns the balance. 
I can remove this information from the dataProps, but I'll have to redo the call to the contract in `getBalances`
Is that ok ?

Reply:
ah okok get it